pkg/license/license.go

```diff
@@ -10,13 +10,65 @@ import (
 	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
 	"github.com/replicatedhq/kots/kotskinds/client/kotsclientset/scheme"
 	"github.com/replicatedhq/kots/pkg/buildversion"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/store"
 )
 
 type LicenseData struct {
 	LicenseBytes []byte
 	License      *kotsv1beta1.License
 }
 
+func ResolveExistingLicense(verifiedLicense *kotsv1beta1.License) (bool, error) {
+	installedAppSet := make(map[string]bool)
+
+	apps, err := store.GetStore().ListInstalledApps()
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to list installed apps"))
+		return false, err
+	}
+
+	notInstalledApps, err := store.GetStore().ListNotInstalledApps()
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to list not-installed apps"))
+		return false, err
+	}
+
+	// build set of installed app slugs using this license
+	for _, app := range apps {
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		obj, _, err := decode([]byte(app.License), nil, nil)
+		if err != nil {
+			continue
+		}
+		license := obj.(*kotsv1beta1.License)
+
+		if license.Spec.LicenseID == verifiedLicense.Spec.LicenseID {
+			installedAppSet[app.Slug] = true
+		}
+	}
+
+	for _, app := range notInstalledApps {
```

Comment:
`notInstalledApps` should already have the license, no need for listing installed apps